### ISSUE TITLE
Multi loss training in keras: how does it work

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#10779]
Good morning. I am trying to set up an adversarial model. One part of the model is defined as follows:

`model = Model(input=input_layer, output=[net1(input_layer), net2(net1(input_layer))])`
`model.compile(loss=[loss1, loss2], loss_weights=[10.0, 1.0], optimizer=optimizer_model)`

So I couldn't really get much from the official documentation, but looking around I saw that someone thinks the two losses are individually minimized, while other people say the sum of the two losses gets minimized during the training.
Now I do get a loss which is the sum of the two in my output. But at a certain point loss2 gets attracted towards a minimum. Since loss1 and loss2 are adversarial, when loss2 tends to decrease, loss1 tends to get bigger. The result is that by the end of the training loss2 is very small, loss1 is quite huge and the sum of the two, which decreases during the first epochs, becomes bigger and bigger. So I don't understand how keras works. Could you please help me here? Thank you very much,
Marco
